refactor(core): static-query schematic should handle abstract classes

devversion opened the pull request:
Queries can not only be accessed within derived classes, but also in
the super class through abstract methods. e.g.

```ts
abstract class BaseClass {

  abstract getEmbeddedForm(): NgForm {}

  ngOnInit() {
     this.getEmbeddedForm().doSomething();
  }
}

class Subclass extends BaseClass {
  @ViewChild(NgForm) form: NgForm;

  getEmbeddedForm() { return this.form; }

}
```

Same applies for abstract properties which are implemented in the base class
through accessors. This case is also now handled by the schematic.

Resolves FW-1213

**NOTE**: Blocked on #29663

kara commented on packages/core/schematics/migrations/static-queries/angular/analyze_query_usage.ts:
```diff
@@ -34,11 +36,15 @@ export function analyzeNgQueryUsage(
 
-/** Checks whether a given class or it's derived classes use the specified query statically. */
+/**
+ * Checks whether a given query is used statically within the given class, it's super
```
```suggestion
 * Checks whether a given query is used statically within the given class, its super
```

devversion replied:
Good catch. fixed

kara commented on packages/core/schematics/migrations/static-queries/angular/analyze_query_usage.ts:
```diff
@@ -74,13 +65,66 @@ function isQueryUsedStatically(
+    return false;
+  }
+
+  // In case derived classes should also be analyzed are classes, we determine the
```
```suggestion
  // In case derived classes that should also be analyzed are classes, we determine the
```

devversion replied:
Good catch. This currently doesn't make any sense. The message should be `In case derived classes should also be analyzed`. fixed. thanks!

kara reviewed (approved):
LGTM, aside from typos

devversion committed:
refactor(core): static-query schematic should handle abstract classes

Queries can not only be accessed within derived classes, but also in
the super class through abstract methods. e.g.

```
abstract class BaseClass {

  abstract getEmbeddedForm(): NgForm {}

  ngOnInit() {
     this.getEmbeddedForm().doSomething();
  }
}

class Subclass extends BaseClass {
  @ViewChild(NgForm) form: NgForm;

  getEmbeddedForm() { return this.form; }

}
```

Same applies for abstract properties which are implemented in the base class
through accessors. This case is also now handled by the schematic.

Resolves FW-1213

kara commented:
[presubmit](http://test/OCL:242510972:BASE:242511018:1554750678177:e83296b)

kara commented:
@devversion Can you rebase this?

devversion commented:
@kara Done.
